pages/dao/[symbol]/proposal/components/instructions/Raydium/RemoveLiquidityFromPool.tsx

```diff
@@ -0,0 +1,157 @@
+import { debounce } from '@utils/debounce'
+import { notify } from '@utils/notifications'
+import {
+  RemoveLiquidityRaydiumForm,
+  UiInstruction,
+} from '@utils/uiTypes/proposalCreationTypes'
+
+import { NewProposalContext } from '../../../new'
+import useInstructionFormBuilder from '@hooks/useInstructionFormBuilder'
+import { removeRaydiumLiquidityPoolSchema } from '../../schemas/validationSchemas'
+import SelectOptionList from '../../SelectOptionList'
+import { GovernedMultiTypeAccount } from '@utils/tokens'
+
+const RemoveLiquidityFromPool = ({
+  index,
+  governanceAccount,
+}: {
+  index: number
+  governanceAccount: GovernedMultiTypeAccount | undefined
+}) => {
+  const {
+    form,
+    connection,
+    formErrors,
+    validateForm,
+    canSerializeInstruction,
+    handleSetForm,
+  } = useInstructionFormBuilder<RemoveLiquidityRaydiumForm>({
+    initialFormValues: {
+      governedAccount: governanceAccount,
+      liquidityPool: '',
+      amountIn: 0,
+    },
+    schema: removeRaydiumLiquidityPoolSchema,
+  })
+
+  const { handleSetInstructions } = useContext(NewProposalContext)
+
+  const [lpMintInfo, setLpMintInfo] = useState<{
+    balance: number
+    decimals: number
+  } | null>(null)
+
+  useEffect(() => {
+    async function fetchLpMintInfo() {
+      try {
+        const { maxBalance, decimals } = await fetchLiquidityPoolData({
+          governanceKey: governanceAccount?.governance.pubkey,
+          lp: form.liquidityPool,
+          connection,
+        })
+        setLpMintInfo({ balance: maxBalance, decimals })
+      } catch (e) {
+        notify({
+          type: 'error',
+          message: 'Could not fetch LP Account',
+          description: `${form.liquidityPool} LP Token Account could not be found for the selected Governance`,
+        })
+      }
+    }
+    fetchLpMintInfo()
+  }, [governanceAccount?.governance.pubkey, form.liquidityPool])
+
+  async function getInstruction(): Promise<UiInstruction> {
+    const isSerializable = await canSerializeInstruction()
+    if (!isSerializable || !form.liquidityPool || !lpMintInfo) {
+      return {
+        serializedInstruction: '',
+        isValid: false,
+        governance: governanceAccount?.governance,
+      }
+    }
+    const createIx = createRemoveLiquidityInstruction(
+      new PublicKey(governanceAccount!.governance.pubkey),
+      jsonInfo2PoolKeys(liquidityPoolKeysList[form.liquidityPool]),
+      new BigNumber(form.amountIn).shiftedBy(lpMintInfo.decimals).toString()
+    )
+
+    return {
+      serializedInstruction: serializeInstructionToBase64(createIx),
+      isValid: true,
+      governance: governanceAccount?.governance,
+    }
+  }
+
+  useEffect(() => {
+    if (form.liquidityPool) {
+      debounce.debounceFcn(async () => {
+        await validateForm()
+      })
+      handleSetForm({
+        value: form.liquidityPool,
+        propertyName: 'liquidityPool',
+      })
+    }
+  }, [form.liquidityPool])
```

Comment:
So when the `form.liquidityPool` changes, we validate the form explicitly (though it's already re-validated automatically?), and then re-sets the value of the `form.liquidityPool` again with the same value?

Reply:
yeah I fucked up, this is fixed with latest refacto